service-base/src/test/java/org/eclipse/hono/service/AbstractProtocolAdapterBaseTest.java

```diff
@@ -202,6 +203,28 @@ public void testGetRegistrationAssertionFailsWith404ForNonExistingDevice(final T
         }));
     }
 
+    /**
+     * Verifies that the adapter fails a request to retrieve a token for a gateway that does not
+     * belong to the same tenant as the device it wants to act on behalf of.
+     * 
+     * @param ctx The vert.x test context.
+     */
+    @Test
+    public void testGetRegistrationAssertionFailsWith403ForNonMatchingTenant(final TestContext ctx) {
+
+        // GIVEN an adapter
+        adapter = newProtocolAdapter(properties, null);
```

Comment:
well, if we would strive for the ability to run test cases in parallel, then I would agree. Since we don't, it doesn't make much of a difference.

Reply:
As long as the adapter is always set like that: yes.
But if a test case would be added where this was forgotten, strange things might happen.

Since this is currently not the case anywhere, no problem - I just think it is a question of test code cleanliness.